src/System.Management.Automation/engine/CommandCompletion/CompletionAnalysis.cs

```diff
@@ -503,6 +503,11 @@ internal List<CompletionResult> GetResultHelper(CompletionContext completionCont
                                 return completions;
                             }
                         }
+                        else if (lastAst.Parent is IndexExpressionAst indexExpressionAst)
```

Comment:
The support for tab completion on index obviously comes in changes in multiple places to cover different scenarios/cases. Can you please add comments at each place to explain what scenario they cover?